OpenRA.Game/GameRules/MusicInfo.cs

```diff
@@ -39,12 +41,18 @@ public void Load()
 				return;
 
 			Exists = true;
-			using (var s = Game.ModData.ModFiles.Open(Filename))
+
+			using (var stream = Game.ModData.ModFiles.Open(Filename))
 			{
-				if (Filename.ToLowerInvariant().EndsWith("wav"))
-					Length = (int)WavLoader.WaveLength(s);
-				else
-					Length = (int)AudLoader.SoundLength(s);
+				foreach (var loader in Game.ModData.SoundLoaders)
```

Comment:
Changed it to expose a `CanParse(Stream)` method via the interface.
The alternative would be to do filename checks, with each parser providing a file extension, but that's not very reliable as it can easily fail on bogus files. (Would be much faster though, so may very well be worth it).